pkg/email/lore/parse.go

```diff
@@ -33,24 +54,128 @@ func Threads(emails []*email.Email) []*Thread {
 	return ctx.threads
 }
 
+// PatchSeries is similar to Threads, but returns only the patch series submitted to the mailing lists.
+func PatchSeries(emails []*email.Email) []*Series {
+	var ret []*Series
+	// Normally, all following series patches are sent in response to the first email sent.
+	// So there's no sense to look at deeper replies.
+	for _, thread := range listThreads(emails, 1) {
+		if thread.Type != dashapi.DiscussionPatch {
+			continue
+		}
+		patch, ok := parsePatchSubject(thread.Subject)
+		if !ok {
+			// It must never be happening.
+			panic("DiscussionPatch is set, but we fail to parse the thread subject")
```

Comment:
Errors processing strategy is not clear to me. How will the panics be monitored? How will you recover after the failures? etc.

Reply:
I'd say this question lies more in scope of the discussion of how to deploy `syz-cluster` and how to organize the error monitoring there rather than in the scope of the discussion of the library package.

Also, here it's `panic()`, which is more about assertions that should never ever be happning rather than about legitimate error processing.

Reply:
Let's better understand how will it work. It'll help to review faster.